Update to post-challenge and new challenge

colbygk opened the pull request:
I was unfamiliar with working against Reddit's APIs, so please feel full freedom to reject or request alterations to this pull request.

What I found was:

- It appears that the praw api has evolved since post-challenges.py was created and I have updated post-challenges.py according to what I believe is the expected behavior.
- As a test, I have added in challenge 336, which post-challenges.py pulled from Reddit.
- In a future pull request, I would like to update the documentation to provide guidance on how to use praw and the setting up of a reddit api access key, as that information was not in a convenient single location.
  - I would also like to update the documentation about the reference to a post_challenges script that seems to no longer exist in the repo and another script referenced inside of post-challeneges.py that is also not there (send-data.sh)
- After we have discussed these updates, I will submit more pull requests that fit what you would prefer.
- The naming convention appears to have shifted and challenges appear in the repo that do not follow what is requested in CONTRIBUTING.md, i.e. #336 vs 0336 and so on.  Would be happy to rename and resubmit those to follow the naming convention

freddiev4 committed:
Merge pull request #180 from Altlock/master

Number two is done

freddiev4 commented on post-challenges.py:
```diff
@@ -27,11 +27,11 @@ def get_current_week(num_challenges=1):
-	chals = sub.get_new(limit=num_challenges)
-	_chals = sub.get_new(limit=num_challenges)
+	chals = sub.new(limit=num_challenges)
+	_chals = sub.new(limit=num_challenges)
```
👍  for update to match the current PRAW API

freddiev4 commented on requirements.txt:
```diff
@@ -0,0 +1,2 @@
+praw>=5.1.0
+pprint>=0.1
```
👍  for adding a `requirements.txt` file.

Can we move this to a `etc/` folder?

freddiev4 commented on post-challenges.py:
```diff
@@ -64,7 +64,7 @@ def get_current_week(num_challenges=1):
 
 	logging.info("Started sending data script")
-	os.system("./send-data.sh")
+#	os.system("./send-data.sh")
```
I think we can just remove this line along with the log above it

freddiev4 reviewed (commented):
Hey @colbygk awesome work. I have some feedback for you below.

In regards to the naming scheme, see if you can run `transform.py` which should end up renaming directories in the proper format.

colbygk committed:
Refactor post_challenges and transform.py

The existing transform looked like it could use some improvement. This seems to work on all of the existing challenges, and also handles dealing with challenges that have unexpected numbers embedded, like a date.

colbygk committed:
Update most recent challenges

colbygk commented:
Hi @freddiev4 I have submitted an update to transform that I think will help it do a better job. I have also backfilled the challenges to fill the gap between 313 and this current week. I just noticed that there appears to be no 334 easy challenge (in the reddit itself)

freddiev4 commented on transform.py:
```diff
@@ -19,57 +19,50 @@ def update_challenge_dir_name(oldName, difficulty):
+    # Found a likely looking number
+    #   tokens now possess' the remaining title
+    #   a_token now has the original and possible multiple iteration number
+    #     such as 166b
```
Just as a style thing, we should probably line up the first word of every line to be in the same column
```
    # Found a likely looking number
    #   tokens now possess' the remaining title
    #   a_token now has the original and possible multiple iteration number
    #     such as 166b
```

Changed to

```
    # Found a likely looking number
    # tokens now possess' the remaining title
    # a_token now has the original and possible multiple iteration number
    # such as 166b
``` 

For every set of comments

freddiev4 commented:
@colbygk just had one comment in regards to style. If you fix that I think we can go ahead and merge this 🙂 great work on refactoring `transform.py`!

colbygk committed:
Update config, dev files, and docs

colbygk commented:
Okay, I think that's it. I did update the README.md that you may want to double check.

colbygk commented:
Okay, fo'reals, I'm done.

freddiev4 commented on README.md:
````diff
@@ -2,18 +2,39 @@
-**post-challenges.py runs on Python 3.x**
+  ```bash
+  $ cd DailyProgrammerChallenges
+  $ mkdir tmp
````
What are we using `tmp/` for here? From what you added above, it looks like we should `cd` into it.

colbygk replied:
Right you are.  Also added a bit about using praw.ini

freddiev4 reviewed (commented):
Added one more comment + you should squash your commits 👍

colbygk commented:
I am unfamiliar with commit squashing.  Recommendation on what to do to squash?

colbygk commented:
Hm, I think I figured it out and squashed the last 5 commits into one.

freddiev4 commented:
@colbygk here's an official guide on [Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) in Git.

The short of it is that you can do:

`git rebase -i @~15` which will allow you to do an interactive rebase, on the head of your branch, up until the last 15 commits.

From there you'll see something like:
```
pick f7f3f6d changed my name a bit
pick 310154e updated README formatting and added blame
pick a5f4a0d added cat-file
```

And you can change everything below the first commit to:
```
pick f7f3f6d changed my name a bit
fixup 310154e updated README formatting and added blame
fixup a5f4a0d added cat-file
```

Which will remove the commit message of every commit but the first (which is 15 commits before the most recent one)

colbygk commented:
Looks like since I used `squash` instead of `fixup` it melded instead of replacing.

freddiev4 commented:
After you squash, I think you'd need to run `git push -f`, as it looks like there are still 17 commits here 🙂 

![image](https://user-images.githubusercontent.com/8339018/31857578-dd93d474-b6af-11e7-991c-b09f74e55e7b.png)

colbygk commented:
Well, now we're down to 11 commits. :)  Were you thinking that it should had more squashing?

freddiev4 commented:
Yeah usually we can squash these down into 2 or 3 commits.

Here I was thinking something like:

```
Update most recent challenges
Refactor post_challenges and transform.py
Update config, dev files, and docs
```

colbygk commented:
Hmm... Okay... maybe done now? Sorry for the thrashing.

freddiev4 commented:
Looks good to me 👍
